### PR TITLE
this commit add solution to task 10 get articleByQueries

### PR DESCRIPTION
**Pull request by Topecoding206**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -315,3 +315,111 @@ describe("GET /api/users", () => {
...
 });
+
+describe("GET /api/articles sort by queries", () => {
+  test("200:should return sorted article created_by", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(11);
+        expect(body.articles).toBeSortedBy("created_at", { descending: true });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:326]: For this test, could we also just check that the returned objects have the topic of 'mitch'?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -315,3 +315,111 @@ describe("GET /api/users", () => {
...
+  });
+  test("400:should return bad request when the query doesnt have valid order property", () => {
+    return request(app)
+      .get("/api/articles?check=asc")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
```
Comment (johnhutcheon) [inline __tests__/app.test.js:444]: Could we just add two more tests here please? One for a 200 where it's a valid query but has no articles (should respond with an empty array), and one for a 404 where it's a non-existent topic query?

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi Tope, great tests so far! Could you just see my comment and make the amends then resubmit please? Thanks!

File: __tests__/app.test.js (hunk 1 above)
Comment (Topecoding206) [reply __tests__/app.test.js:326]: okay

File: __tests__/app.test.js (hunk 2 above)
Comment (Topecoding206) [reply __tests__/app.test.js:444]: okay

**Review by johnhutcheon — APPROVED** [review]
